### PR TITLE
Adds example of primary button with icon

### PR DESCRIPTION
**Pull request by mike-morrison**
Adds @dzearing example for adding an inline button.  I think once the styles are enhanced to take an icon universally, we should leave this example up to show the flexibility of the button.

Side note: @dzearing Can you tell I was hitting the refresh button nonstop waiting for you to open this project up?  :+1:   Great work to you and team.  Can't wait to dig in more!

### CODE REVIEW

File: src/demo/pages/ButtonPage/examples/Button.Basic.Example.scss [hunk 1]
```diff
@@ -1,3 +1,8 @@
 .ms-BasicButtonsExample .ms-Button {
   margin: 10px 0;
 }
+
+.ms-Icon {
```
Comment (dzearing) [inline src/demo/pages/ButtonPage/examples/Button.Basic.Example.scss]: scope this within the example so you don't create a rule that affects all icons

File: src/demo/pages/ButtonPage/examples/Button.Basic.Example.scss [hunk 2]
```diff
@@ -1,3 +1,8 @@
...
+  margin-right: 5px;
+  margin-left: -10px;
```
Comment (dzearing) [inline src/demo/pages/ButtonPage/examples/Button.Basic.Example.scss]: use the $include margin-left(...) macros for RTL sensitive rules. (just search around for $include examples, you see)

File: src/demo/pages/ButtonPage/examples/Button.Basic.Example.tsx [hunk 3]
```diff
@@ -44,6 +44,12 @@ export class ButtonBasicExample extends React.Component<any, IBasicButtonsExampl
         <Label>Command button</Label>
         <Button disabled={ disabled } buttonType={ ButtonType.command } icon='personAdd' description='Description of the action this button takes'>Create account</Button>
 
+        <Label>Primary Button with Icon</Label>
+        <Button disabled={ disabled } buttonType={ ButtonType.primary } title='Star' description='Description of the action this button takes'>
```
Comment (dzearing) [inline src/demo/pages/ButtonPage/examples/Button.Basic.Example.tsx]: you could add a classname to this button if you'd like, e.g. 'ms-ButtonBasicExample-customButton'

then your scoped css would be:

.ms-ButtonBasicExample-customButton .ms-Icon {
...
}

**Comment by dzearing** [comment]
Haha :) Thank you!

**Comment by mike-morrison** [comment]
I am going to watch the PluralSight course on Sass.  I will update the PR shortly.  :)

**Comment by mike-morrison** [comment]
@dzearing added the requested changes.

**Comment by dzearing** [comment]
Closing this PR, as I think we don't want to take this one for now.

**Pull request closed without merging**